sympy/series/tests/test_formal.py

```diff
@@ -507,3 +507,10 @@ def test_fps__operations():
     fi = f2.integrate(x)
     assert fi.function == sin(x)
     assert fi.truncate() == x - x**3/6 + x**5/120 + O(x**6)
+
+def test_fps__convolutions():
+    f1, f2, f3 = fps(sin(x)), fps(exp(x)), fps(cos(x))
+
+    assert f1.convolve(f2, x, 6) == x + x**2 + x**3/3 - x**5/12 + O(x**6)
+    assert f1.convolve(f2, x, 7) == x + x**2 + x**3/3 - x**5/12 - x**6/36 + O(x**7)
+    assert f1.convolve(f3, x, 7) == x - 2*x**3/3 + x**5/12 + O(x**7)
```

Comment:
Add tests when say ``x0`` is not 0.

What happens in the case:

```
f1, f2 = fps(sin(x), fps(exp(y))

f1.convolute(f2, x)
```
Also, add simple tests for raising exception when ``dir`` or ``x0`` are not same.

Reply:
This is a case of multivariate formal power series. I think we can raise an error, if `self.x` is not equal to `other.x`. I will add tests for these exceptions.